pkg/dwarf/reader/reader.go

```diff
@@ -319,3 +319,50 @@ func (reader *Reader) NextCompileUnit() (*dwarf.Entry, error) {
 
 	return nil, nil
 }
+
+type Entry interface {
+	Val(dwarf.Attr) interface{}
+}
+
+type compositeEntry []*dwarf.Entry
+
+func (ce compositeEntry) Val(attr dwarf.Attr) interface{} {
+	for _, e := range ce {
+		if r := e.Val(attr); r != nil {
+			return r
+		}
+	}
+	return nil
+}
+
+// LoadAbstractOrigin loads the entry corresponding to the
+// DW_AT_abstract_origin of entry and returns a combination of entry and its
+// abstract origin.
+func LoadAbstractOrigin(entry *dwarf.Entry, dwarfData *dwarf.Data, aordr **dwarf.Reader) (Entry, dwarf.Offset) {
+	ao, ok := entry.Val(dwarf.AttrAbstractOrigin).(dwarf.Offset)
+	if !ok {
+		return entry, entry.Offset
+	}
+
+	if *aordr == nil {
+		*aordr = dwarfData.Reader()
```

Comment:
I was trying to avoid allocating a second reader every time since we'll almost never use it.

Reply:
I get that, I would just like to clean up this signature a bit and make intent clearer. What do you think about passing in the scope directly instead of the separate readers?

Reply:
I'm ok with doing that but it needs to be moved to pkg/proc.

Reply:
Ah, good point, circular import right? Not a fan of moving this into `pkg/proc`. 

Why not just allocate and store a reader in `BinInfo`? We can always seek to where we need to be, I can't remember off the top of my head if this will mess with any optimizations we have in place. If so, maybe we can store a reader on scope and use that instead of specifically calling it an `aoreader`. Perhaps we could memoize the call to get a new reader when attached to a specific scope or instance of `BinaryInfo`.